plugins/systemadmin/systemadmin.plugin.zsh

```diff
@@ -140,12 +140,16 @@ d0() {
 
 # gather external ip address
 geteip() {
-    curl http://ifconfig.me
+    echo "$(curl -s -S http://ipecho.net/plain)"
 }
 
 # determine local IP address
 getip() {
-    ifconfig  | grep 'inet addr:'| grep -v '127.0.0.1' | cut -d: -f2 | awk '{ print $1}'
+    if [ $(hash | grep '^ip=') ]; then
```

Comment:
```hash``` is quite similar to ```which```, except is just lists all the executables known in the current $PATH, one per line. In this case I preferred it instead of ```which``` because some implementations of the latter produce output even if the program passed as a parameter is not found.